Core/Inc/monitor.h

```diff
@@ -8,30 +8,43 @@
 #include "pdu.h"
 #include "steeringio.h"
 
-typedef struct {
-	uint16_t brake_value;
-	uint16_t accelerator_value; /* 0-100 */
-} pedals_t;
+/**
+ * @brief Get the debounced TSMS reading.
+ * 
+ * @return State of TSMS.
+ */
+bool get_tsms();
 
-pedals_t get_pedal_data();
-bool get_brake_state();
+typedef struct {
+	mpu_t *mpu;
+	pdu_t *pdu;
+} non_func_data_args_t;
 
-bool get_tsms_reading();
+/**
+ * @brief Task for collecting non functional data from the car. Includes LV battery voltage reading and fuse monitor reading.
+ * 
+ * @param pv_params Pointer to non_func_data_args_t
+ */
+void vNonFunctionalDataCollection(void *pv_params);
+extern osThreadId_t non_functional_data_thead;
+extern const osThreadAttr_t non_functional_data_attributes;
 
-/* Defining Temperature Monitor Task */
-void vTempMonitor(void *pv_params);
-extern osThreadId_t temp_monitor_handle;
-extern const osThreadAttr_t temp_monitor_attributes;
+/* Arguments for the data collection thread */
+typedef struct {
+	pdu_t *pdu;
+	steeringio_t *wheel;
+} data_collection_args_t;
 
-/* Task for Petting the Watchdog */
-void vWatchdogMonitor(void *pv_params);
-extern osThreadId_t watchdog_monitor_handle;
-extern const osThreadAttr_t watchdog_monitor_attributes;
+/**
+ * @brief Task for collecting functional data from the car. Includes TSMS and steering wheel button monitoring.
+ * 
+ * @param pv_params Pointer to data_collection_args_t
+ */
+void vDataCollection(void *pv_params);
+extern osThreadId_t data_collection_thread;
+extern const osThreadAttr_t data_collection_attributes;
 
-/* Task for Reading in Pedal Inputs (Brakes + Accelerator) */
-void vPedalsMonitor(void *pv_params);
-extern osThreadId_t pedals_monitor_handle;
-extern const osThreadAttr_t pedals_monitor_attributes;
+/* Unused -------------------------------------------------------------------------------------------------------*/
 
 /* Task for Monitoring the IMU */
 void vIMUMonitor(void *pv_params);
```

Comment:
same as last review i think but just not sure why this needs own thread but whatevs

Reply:
I'm leaving it alone because it is not used. We will come back to address this eventually.